configuration/pih/htmlforms/hiv/section-hiv-state.xml

```diff
@@ -156,10 +156,248 @@
+        const apiBaseUrl =  "/" + contextPath + "/ws/rest/v1";
+        const patientUuid = '<lookup expression="patient.uuid"/>';
+
+        let currentEncounterDate = new Date();
+        let encounterDate = '<lookup expression="encounter.getEncounterDatetime().getTime()"/>';
+        if (typeof encounterDate !== "undefined" &amp;&amp; encounterDate !== null &amp;&amp; (encounterDate.length > 0)) {
+            currentEncounterDate = new Date(+encounterDate);
+        }
+
+        let treatmentLineConceptUuid="4dfefa8d-9940-4ad5-800d-f16579b37d54";
+
+         let lastRecordedTreatmenLine = {
+            treatmenLine: "",
+            date: null,
+            encounterDisplay: ""
+        };
+
+        
+           jq.getJSON(apiBaseUrl + "/obs", {
+            concept: treatmentLineConceptUuid,
+            patient: patientUuid,
+            v: 'custom:(uuid,display,obsDatetime,value,encounter:(display,encounterDatetime,location:(display),encounterType:(display)))'
+          },
+          function( data ){
+            if (data.results.length > 0) {
+                for (let index = 0; index &lt; data.results.length; ++index) {
+                    let treatmentLineObs = data.results[index];
+                    let treatmentLineObsDate = new Date(treatmentLineObs.obsDatetime);
+                    let tmpTreatmentLineObs = {
+                      treatmenLine: "",
+                      date: null,
+                      encounterDisplay: ""
+                    };
+
+                    if( (currentEncounterDate.getTime() >=  treatmentLineObsDate.getTime()) ||
+                        (currentEncounterDate.toDateString() === treatmentLineObsDate.toDateString())) {
+                            if (typeof treatmentLineObs.value !== "undefined" &amp;&amp; treatmentLineObs.value !== null) {
+                                tmpTreatmentLineObs.treatmenLine = treatmentLineObs.value;
+                                tmpTreatmentLineObs.date = treatmentLineObsDate;
+                                tmpTreatmentLineObs.encounterDisplay = treatmentLineObs.encounter.display;
+                    }
+                    }
+                    if ( (tmpTreatmentLineObs.date != null) &amp;&amp;
+                        ((lastRecordedTreatmenLine.date == null) || (lastRecordedTreatmenLine.date.getTime() &lt; tmpTreatmentLineObs.date.getTime()) ) ) {
+
+                        lastRecordedTreatmenLine.treatmenLine = tmpTreatmentLineObs.treatmenLine;
+                        lastRecordedTreatmenLine.date = tmpTreatmentLineObs.date;
+                        lastRecordedTreatmenLine.encounterDisplay = tmpTreatmentLineObs.encounterDisplay;
+                    }
+                }
+            }
+
+           
+                if (lastRecordedTreatmenLine.date != null) {
+                  
+                   if(lastRecordedTreatmenLine.treatmenLine.display !==null){
+
+                      setValue('Isoniazide.value', 1505);
```

Comment:
Is 1505 the concept id?  If so, unfortunately, the concept ids aren't consistent across servers, so we'd likely need to do something a little more complex where we look up the concept id based on the uuid or mapping of the concept.

Reply:
I assume this can be done via a velocity expression to look up a concept by mapping or uuid.  I also started writing some javascript services that can be used / added to for utlities like this:  https://github.com/PIH/openmrs-config-pihemr/blob/master/configuration/pih/scripts/global/pihServices.js#L33